source/boundary_traction/interface.cc

```diff
@@ -59,8 +59,375 @@ namespace aspect
+                                                + error));
+              }
+
+            if (boundary_traction_indicators.find(boundary_id) != boundary_velocity_indicators.end())
+              {
+                Assert(boundary_tractiony_indicators[boundary_id].first == comp,
+                       ExcMessage("Different traction plugins for the same boundary have to have the same component selector. "
+                                  "This was not the case for boundary: " + key_and_comp +
+                                  ", for plugin: " + value + ", with component selector: " + comp));
+
+                // finally, put it into the list
+                boundary_traction_indicators[boundary_id].second.push_back(value);
+              }
+            else
+              {
+                boundary_traction_indicators[boundary_id] = std::make_pair(comp,std::vector<std::string>(1,value));
+              }
+          }
+      }
+      prm.leave_subsection();
+
+      // go through the list, create objects and let them parse
+      // their own parameters
+      for (const auto &boundary_id : boundary_traction_indicators)
+        {
+          for (const auto &name : boundary_id.second.second)
+            {
+              boundary_traction_objects[boundary_id.first].push_back(
+                std::unique_ptr<Interface<dim>> (std::get<dim>(registered_plugins)
+                                                  .create_plugin (name,
+                                                                  "Boundary traction::Model names")));
+
+              if (SimulatorAccess<dim> *sim = dynamic_cast<SimulatorAccess<dim>*>(boundary_traction_objects[boundary_id.first].back().get()))
+                sim->initialize_simulator (this->get_simulator());
+
+              boundary_traction_objects[boundary_id.first].back()->parse_parameters (prm);
+              boundary_traction_objects[boundary_id.first].back()->initialize ();
+            }
+        }
+    }
+      template <int dim>
+    void
+    Manager<dim>::write_plugin_graph (std::ostream &out)
+    {
+      std::get<dim>(registered_plugins).write_plugin_graph ("Boundary traction interface",
+                                                            out);
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace internal
+  {
+    namespace Plugins
+    {
+      template <>
+      std::list<internal::Plugins::PluginList<BoundaryTraction::Interface<2>>::PluginInfo> *
+      internal::Plugins::PluginList<BoundaryTraction::Interface<2>>::plugins = nullptr;
+      template <>
+      std::list<internal::Plugins::PluginList<BoundaryTraction::Interface<3>>::PluginInfo> *
+      internal::Plugins::PluginList<BoundaryTraction::Interface<3>>::plugins = nullptr;
+    }
+  }
+
+  namespace BoundaryTraction
+  {
+#define INSTANTIATE(dim) \
+  template class Interface<dim>; \
+  template class Manager<dim>;
+
+    ASPECT_INSTANTIATE(INSTANTIATE)
+
+#undef INSTANTIATE
```

Comment:
There is already a section further down in this file labeled `#explicit instantiations`. Give it a try to merge your changes here with that section further down and find me if you run into problems.